server/sonar-server/src/main/java/org/sonar/server/usergroups/ws/DeleteAction.java

```diff
@@ -91,34 +89,36 @@ public void handle(Request request, Response response) throws Exception {
     }
   }
 
-  private void checkNotTryingToDeleteDefaultGroup(DbSession dbSession, long groupId) {
+  private void checkNotTryingToDeleteDefaultGroup(DbSession dbSession, GroupId group) {
+    // TODO support organizations
     String defaultGroupName = settings.getString(CoreProperties.CORE_DEFAULT_GROUP);
     GroupDto defaultGroup = dbClient.groupDao().selectOrFailByName(dbSession, defaultGroupName);
-    checkArgument(groupId != defaultGroup.getId(),
+    checkArgument(group.getId() != defaultGroup.getId().longValue(),
       format("Default group '%s' cannot be deleted", defaultGroupName));
   }
 
-  private void checkNotTryingToDeleteLastSystemAdminGroup(DbSession dbSession, String groupName) {
-    boolean hasAdminPermission = dbClient.roleDao()
-      .selectGroupPermissions(dbSession, groupName, null)
+  private void checkNotTryingToDeleteLastSystemAdminGroup(DbSession dbSession, GroupId group) {
+    boolean hasAdminPermission = dbClient.groupPermissionDao()
+      .selectGroupPermissions(dbSession, group.getId(), null)
       .contains(GlobalPermissions.SYSTEM_ADMIN);
+    // TODO support organizations
     boolean isOneRemainingAdminGroup = dbClient.groupPermissionDao().countGroups(dbSession, GlobalPermissions.SYSTEM_ADMIN, null) == 1;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "countGroups"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)